src/main/java/DilemmaDetector/Simulator/CollisionDetector.java

```diff
@@ -49,14 +58,57 @@ public boolean detectOutOfRoad(Actor mainVehicle){
+                && distanceBetweenRigidBodies.x < (surroundingLength + mainVehicle.getRigidBody().getLength()) /2) {
+            //small change x and y
+            System.out.println("COLL WITH SURROUNDING: " );
+            System.out.println("MAIN VEHICLE POS: " + mainVehicle.getRigidBody().getPosition());
+            System.out.println("MAIN VEHICLE SHAPE: "  + mainVehicle.getRigidBody().getWidth() +
+                    " " + mainVehicle.getRigidBody().getLength());
+            System.out.println("SURROUNDING POS: " + surrounding.getPosition());
+            System.out.println("OTHER VEHICLE SHAPE: "  + surroundingWidth +
+                    " " + surroundingLength);
+            System.out.println("DISTANCE  " + distanceBetweenRigidBodies + " " + distanceBetweenRigidBodies.x +  " " + distanceBetweenRigidBodies.y);
+            System.out.println("COUNTED : " + (surroundingWidth + mainVehicle.getRigidBody().getWidth()) /2 + "  " +
+                    (surroundingLength + mainVehicle.getRigidBody().getLength()) /2);
+
+            System.out.println(mainVehicle.getRigidBody().getPosition());
+
+            isCollision = true;
+        }
+        return isCollision;
+    }
+
+
 
     public boolean detectCollisionWithRigidBodyInMoment(RigidBody rigidBody) {
         boolean isCollision = false;
         double vehicleWidth = rigidBody.getWidth();
         double vehicleLength = rigidBody.getLength();
         Vector2 distanceBetweenRigidBodies = getDistanceBetweenRigidBodies(mainVehicle.getRigidBody(), rigidBody);
-        if(distanceBetweenRigidBodies.x < (vehicleWidth + mainVehicle.getRigidBody().getWidth()) /2
-                && distanceBetweenRigidBodies.y < (vehicleLength + mainVehicle.getRigidBody().getLength()) /2) {
+        if(distanceBetweenRigidBodies.y < (vehicleWidth + mainVehicle.getRigidBody().getWidth()) /2
+                && distanceBetweenRigidBodies.x < (vehicleLength + mainVehicle.getRigidBody().getLength()) /2) {
+        //small change x and y
+            System.out.println("DETECTED COLL" );
+            System.out.println("MAIN VEHICLE POS: " + mainVehicle.getRigidBody().getPosition());
+            System.out.println("MAIN VEHICLE SHAPE: "  + mainVehicle.getRigidBody().getWidth() +
+                    " " + mainVehicle.getRigidBody().getLength());
+            System.out.println("OTHER VEHICLE POS: " + rigidBody.getPosition());
+            System.out.println("OTHER VEHICLE SHAPE: "  + vehicleWidth +
+                    " " + vehicleLength);
+            System.out.println("DISTANCE  " + distanceBetweenRigidBodies + " " + distanceBetweenRigidBodies.x +  " " + distanceBetweenRigidBodies.y);
+            System.out.println("COUNTED : " + (vehicleWidth + mainVehicle.getRigidBody().getWidth()) /2 + "  " +
+            (vehicleLength + mainVehicle.getRigidBody().getLength()) /2);
+
+
+
+            System.out.println(mainVehicle.getRigidBody().getPosition());
+
```

Comment:
Powtórzony kod, tylko print się różni, wydziel osobną funkcję, która będzie robić to co te 2 z parametrem string co obsługujesz (SURROUNDING / OTHER VEHICLE / dokładna nazwa aktora), też to nie będzie zawsze other vehicle, może być np zwierze.